lib/src/main/java/com/diffplug/spotless/generic/NativeCmdStep.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.generic;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.ProcessRunner;
+
+public class NativeCmdStep {
+	// prevent direct instantiation
+	private NativeCmdStep() {}
+
+	public static FormatterStep create(String name, String pathToExe, List<String> arguments) {
+		Objects.requireNonNull(name, "name");
+		Objects.requireNonNull(pathToExe, "pathToExe");
+		List<String> argumentsWithPathToExe = new ArrayList<>();
+		argumentsWithPathToExe.add(pathToExe);
+		if (arguments != null) {
+			argumentsWithPathToExe.addAll(arguments);
+		}
+		return FormatterStep.createLazy(name, () -> new State(pathToExe, argumentsWithPathToExe), State::toFunc);
+	}
+
+	static class State implements Serializable {
+		private static final long serialVersionUID = 1L;
+
+		final String pathToExe;
```

Comment:
I think you're going to want `FileSignature` of the exe. That's not perfect, because the exe might be calling out to .dlls that change, but it's better than just the path. It's also machine-independent, which will let gradle use remote buildcache.